### DISCUSSION TITLE
[Azure Devops agent] Automatic Installation

### DISCUSSION THREAD

**Discussion by nico3133** [actions/virtual-environments#8735] [General questions] [answered]
Hello, 

Just managed to build a template and a VM with the scripts, and everything is fine, but just one question, is it a way to automatize the installation of an Azure Agent, with the devops orga parameters and PAT, to fully automatize the deployment ?
Maybe it's already done in the project, but I had just checked the different scripts and I don't find such thing .

Many thanks for your work and your help

Nicolas

**Comment by conradj3** [comment, marked as answer]
Hey Dude use Azure DevOps Scaleset Agents. https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops

**Reply by nico3133** [reply]
Hey @conradj3 thanks for the answer, I will try that !
Good idea

**Reply by wortner** [reply]
What if I need to install the agent myself? If I understand it correctly the VMSS is "injected" with the Agent binary/setup by the Azure Devops. It would be great to have possibility to use the concept of runner-images toolset, install it ADO VSTS agent, runconfig.sh with appropriate ADO auth method and install as a service. 

Only other option for me seems to be running install script after packer build on running VM, or invoke Agent creation ADO REST API call with autoProvision to true.

**Reply by nico3133** [reply]
Just tested with the VMSS and it works perfectly, it installed automatically an agent into created VM by VMSS. It's a kind of magic ^^
You just need to create the vmss on you azure sub, make a service connection, and add a new agent pool on your devops and you select VMSS agent pool. Of course you need a proper image file created with the scripts

**Reply by wortner** [reply]
This is complete manual process, this way the pool creation cannot be automated. @conradj3 Any hints for my part?

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
